lib/monitoring/middleware/prometheus_collector.rb

```diff
@@ -0,0 +1,52 @@
+require 'benchmark'
+
+module Monitoring
+  module Middleware
+    class PrometheusCollector
+      attr_reader :app
+
+      def initialize(app)
+        @app = app
+      end
+
+      def call(env) # :nodoc:
+        trace(env) { @app.call(env) }
+      end
+
+      protected
+
+      # Trace HTTP requests
+      def trace(env)
+        response = nil
+        duration = Benchmark.realtime { response = yield }
+        record(env, response.first.to_s, duration)
+        return response
+      rescue => exception
+        Monitoring::PubSub.instance.publish(
+          "conjur.request_exception", 
+          exception: exception
+        )
+        raise
+      end
+
+      def record(env, code, duration)
+        Monitoring::PubSub.instance.publish(
```

Comment:
Thinking it might good dependency inject the pubsub. It would make unit testing a bit simpler.